source/common/ssl/ssl_socket.h

```diff
@@ -123,7 +125,8 @@ class ServerSslSocketFactory : public Network::TransportSocketFactory,
   SslSocketFactoryStats stats_;
   ServerContextConfigPtr config_;
   const std::vector<std::string> server_names_;
-  ServerContextSharedPtr ssl_ctx_;
+  mutable absl::Mutex ssl_ctx_mu_;
```

Comment:
It seems that Envoy is using std::shared_timed_mutex.   What is the reason not using it here?

Reply:
using absl::Mutex allows us to use `GUARDED_BY(mutex)` and other thread annotations, and Envoy uses `absl::Mutex` (or a wrapper, `Thread::MutexBasicLockable`) almost everywhere.